internal/uvm/wait.go

```diff
@@ -3,17 +3,34 @@
 package uvm
 
 import (
-	"github.com/Microsoft/hcsshim/internal/logfields"
+	"context"
+	"fmt"
+
 	"github.com/sirupsen/logrus"
+
+	"github.com/Microsoft/hcsshim/internal/logfields"
 )
 
 // Wait waits synchronously for a utility VM to terminate.
-func (uvm *UtilityVM) Wait() error {
-	err := uvm.hcsSystem.Wait()
+func (uvm *UtilityVM) Wait() error { return uvm.WaitCtx(context.Background()) }
+
+// Wait waits synchronously for a utility VM to terminate, or the context to be cancelled.
+func (uvm *UtilityVM) WaitCtx(ctx context.Context) (err error) {
+	err = uvm.hcsSystem.WaitCtx(ctx)
 
-	logrus.WithField(logfields.UVMID, uvm.id).Debug("uvm exited, waiting for output processing to complete")
+	e := logrus.WithField(logfields.UVMID, uvm.id)
+	e.Debug("uvm exited, waiting for output processing to complete")
 	if uvm.outputProcessingDone != nil {
-		<-uvm.outputProcessingDone
+		select {
+		case <-uvm.outputProcessingDone:
+		case <-ctx.Done():
+			if err2 := ctx.Err(); err2 != nil {
```

Comment:
We don't need to check `err2` for nilness explicitly, as it is guaranteed for it to be non-nil if `ctx.Done()` is closed.

https://pkg.go.dev/context#Context.Err